lib/ProgressPlugin.js

```diff
@@ -118,212 +120,233 @@ class ProgressPlugin {
 		this.showActiveModules = options.activeModules;
 	}
 
+	/**
+	 * @param {Compiler | MultiCompiler} compiler webpack compiler
+	 * @returns {void}
+	 */
 	apply(compiler) {
-		const { modulesCount } = this;
 		const handler = this.handler || createDefaultHandler(this.profile);
+		if (compiler instanceof MultiCompiler) {
+			this._applyOnMultiCompiler(compiler, handler);
+		} else if (compiler instanceof Compiler) {
```

Comment:
This is test is required because TypeScript is not smart enough to figure out that `compiler` is a `Compiler`.